### PR TITLE
Missing tests

### PR DESCRIPTION
**Pull request by tomato42**
### Description

add tests that are missing in the `tlslite-ng.json` files to it, fix minor issues that prevented them being added there before

### Motivation and Context

ensuring that the tests are runnable as tlsfuzzer develops

### Checklist

- [x] I have read the [CONTRIBUTING.md](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md) document and my PR follows [change requirements](https://github.com/tomato42/tlsfuzzer/blob/master/CONTRIBUTING.md#change-requirements) therein
- [x] the changes are also reflected in documentation and code comments
- [ ] all new and existing tests pass (see Travis CI results)
- [x] [test script checklist](https://github.com/tomato42/tlsfuzzer/wiki/Test-script-checklist) was followed for new scripts
- [x] new test script added to `tlslite-ng.json` and `tlslite-ng-random-subset.json`
- [ ] new and modified scripts were ran against popular TLS implementations:
  - [ ] OpenSSL
  - [ ] NSS
  - [ ] GnuTLS
- [x] required version of tlslite-ng updated in requirements.txt and README.md

---
This change is [Reviewable](https://reviewable.io/reviews/tomato42/tlsfuzzer/415)

### CODE REVIEW

File: tests/tlslite-ng.json [hunk 1]
```diff
@@ -103,6 +103,7 @@
          {"name" : "test-fallback-scsv.py",
           "arguments" : ["--tls-1.3"]
          },
+         {"name" : "test-ffdhe-negotiation.py"},
          {"name" : "test-fuzzed-ciphertext.py"},
          {"name" : "test-fuzzed-finished.py"},
          {"name" : "test-fuzzed-MAC.py"},
```
Comment (nmav) [inline tests/tlslite-ng.json]: r+

File: tlsfuzzer/helpers.py [hunk 2]
```diff
@@ -151,3 +151,19 @@ def psk_ext_updater(psk_settings):
     type and the 3-byte length), but excluding other binders.
     """
     return partial(_psk_ext_updater, psk_settings=psk_settings)
+
+
+def flexible_getattr(val, val_type):
+    """Convert a sring of number, name, or None to object.
```
Comment (nmav) [inline tlsfuzzer/helpers.py]: typo: sring

File: tests/test_tlsfuzzer_helpers.py [hunk 3]
```diff
@@ -13,7 +13,8 @@
     from unittest.mock import call
```
Comment (nmav) [inline tests/test_tlsfuzzer_helpers.py]: The commit text is quite silent on why is that needed. The next commit clarifies it though.

File: tests/tlslite-ng.json [hunk 4]
```diff
@@ -148,6 +148,11 @@
          {"name" : "test-sessionID-resumption.py"},
          {"name" : "test-sig-algs.py"},
          {"name" : "test-signature-algorithms.py"},
+         {"name" : "test-ssl-death-alert.py",
+          "arguments" : ["-n", "0",
+                         "--alert-level", "warning",
+                         "--alert-description", "close_notify"],
+          "comment" : "tlslite-ng does not allow client to send warning alerts, thus the numer of allowed alerts (-n) is 0"},
          {"name" : "test-sslv2-connection.py"},
          {"name" : "test-sslv2-force-cipher-3des.py"},
```
Comment (nmav) [inline tests/tlslite-ng.json]: r+

**Comment by tomato42** [comment]
@nmav updated
